ecs-cli/modules/utils/regcreds/creds_input_reader.go

```diff
@@ -0,0 +1,95 @@
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package readers
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// ECSRegCredsInput contains registry cred entries for creation and/or use in a task execution role
+type ECSRegCredsInput struct {
+	Version             string
+	RegistryCredentials RegistryCreds `yaml:"registry_credentials"`
+}
+
+// RegistryCreds is a map of registry names to RegCredEntry structs
+type RegistryCreds map[string]RegistryCredEntry
+
+// RegistryCredEntry contains info needed to create an AWS Secrets Manager secret and match it to an ECS container(s)
+type RegistryCredEntry struct {
+	SecretManagerARN string   `yaml:"secret_manager_arn"`
+	Username         string   `yaml:"username"`
+	Password         string   `yaml:"password"`
+	KmdKeyID         string   `yaml:"kms_key_id"`
+	ContainerNames   []string `yaml:"container_names"`
+}
+
+// ReadCredsInput parses 'registry-creds up' input into an ECSRegCredsInput struct
+func ReadCredsInput(filename string) (*ECSRegCredsInput, error) {
+	if filename == "" {
+		return nil, errors.New("file is required")
+	}
+
+	rawCredsInput, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error reading file '%v'", filename)
+	}
+	credsInput := &ECSRegCredsInput{}
+
+	if err = yaml.Unmarshal([]byte(rawCredsInput), &credsInput); err != nil {
+		return nil, errors.Wrapf(err, "Error unmarshalling yaml data from credential input file: %s", filename)
+	}
+
+	expandedCredsInput := RegistryCreds{}
+	for regName, credEntry := range credsInput.RegistryCredentials {
+		expandedCredEntry := expandCredEntry(credEntry)
+		expandedCredsInput[regName] = expandedCredEntry
+	}
+
+	credsInput.RegistryCredentials = expandedCredsInput
+
+	return credsInput, nil
+}
+
+// expandCredEntry checks if individual fields are env vars and if so, retrieves & sets that value
+func expandCredEntry(credEntry RegistryCredEntry) RegistryCredEntry {
+	expandedSecretARN := getValueOrEnvVar(credEntry.SecretManagerARN)
+	expandedUsername := getValueOrEnvVar(credEntry.Username)
+	expandedPassword := getValueOrEnvVar(credEntry.Password)
+	expandedKmsKeyID := getValueOrEnvVar(credEntry.KmdKeyID)
+	//TODO: look for env vars in container names?
+
+	expandedCredEntry := RegistryCredEntry{
+		SecretManagerARN: expandedSecretARN,
+		Username:         expandedUsername,
+		Password:         expandedPassword,
+		KmdKeyID:         expandedKmsKeyID,
+		ContainerNames:   credEntry.ContainerNames,
+	}
+	return expandedCredEntry
+}
+
+// selectively runs ExpandEnv() to avoid indescriminant replacement of substrings with '$'
+// e.g., password='c00l$tuff2018' -> return same; password='${MY_PASSWORD}' -> return env value.
+func getValueOrEnvVar(s string) string {
+	if strings.HasPrefix(s, "${") && strings.HasSuffix(s, "}") {
```

Comment:
Hmm, I think it might be okay if we only support `${...}` format for env vars in unixy systems, and not support cases where you have something like `$PASSWORD` because that's more ambiguous and your password might start with a $. But, for the sake of Windows, this should probably support `%PASSWORD%`

Reply:
Ooh. That's a good point. We may have to account for different os's in the tests for this sort of thing if we're relying on go to make a system call for this.

Reply:
I thought about this a bit (as a Windows user myself :P). 

`os.ExpandEnv()` does not work with `%WinVarFormat%`, so we'd need a special case that trims the provided value and calls `os.Getenv()` - which is doable, I suppose, but we also don't support `%ThisFormat%` for env vars in any other file we take in (`compose` , `ecs-parms`), so I was inclined to skip the extra work need to make it work since there's not currently a precedence for it. 

However, one could argue this is a whole other cmd/domain and so we shouldn't be beholden to conventions in other parts of the tool. I respect that reasoning, but I also don't think it's a stretch to maintain that we recommend `${FORMAT}` whenever using env vars in the ECS CLI. So I guess I'm ambivalent about adding it? If you're still inclined to add it despite the above reasoning, let me know and I can build it out.

I agree with your first point (this was my reasoning for not supporting `$PASSWORD`).

Reply:
> but I also don't think it's a stretch to maintain that we recommend ${FORMAT} whenever using env vars in the ECS CLI.

Wait are you saying that the `${ENV_VAR}` format with `os.ExpandEnv` works on windows?

Reply:
Yes! Or at least it worked for me when I ran this code on my Windows machine (see above test output, which uses `${USERNAME}` in the input file but results in the local env var* being read into ECS CLI)

*shown as [REDACTED] in cmd output.